app/database/operator/server_data_operator/handlers/thread.ts

```diff
@@ -105,10 +106,12 @@ const ThreadHandler = (superclass: any) => class extends superclass {
     handleThreadParticipants = async ({threadsParticipants, prepareRecordsOnly, skipSync = false}: HandleThreadParticipantsArgs): Promise<ThreadParticipantModel[]> => {
         const batchRecords: ThreadParticipantModel[] = [];
 
-        if (!threadsParticipants.length) {
-            throw new DataOperatorException(
-                'An empty "thread participants" array has been passed to the handleThreadParticipants method',
+        if (!threadsParticipants?.length) {
+            // eslint-disable-next-line no-console
+            console.warn(
+                'An empty or undefined "threadParticipants" array has been passed to the handleThreadParticipants method',
             );
+            return [];
```

Comment:
```suggestion
           return Promise.resolve([]);
```

Reply:
Async function, return type is already a Promise